### DISCUSSION TITLE
ndk_arm64 error

### DISCUSSION THREAD

**Discussion by soso0726cn** [rustdesk/rustdesk#8686] [Q&A] [answered]
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.6 LTS
Release:	20.04
Codename:	focal

I want to build Android apk with flutter 
but when i run flutter/ndk_arm64.sh

error: 
  = note: ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(vpx_decoder.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(vpx_encoder.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(vpx_codec.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(vpx_image.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(vp8_cx_iface.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(vp8_dx_iface.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(vp9_cx_iface.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(vp9_dx_iface.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(aom_codec.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(aom_decoder.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(aom_encoder.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(aom_image.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(av1_dx_iface.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(av1_cx_iface.c.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(convert.cc.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(convert_argb.cc.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libscrap-815d4982efef5d2c.rlib(convert_from_argb.cc.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libmagnum_opus-74c3b5e4640ff11a.rlib(celt.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libmagnum_opus-74c3b5e4640ff11a.rlib(opus.o) is incompatible with aarch64linux
          ld: error: /app/rustdesk/target/aarch64-linux-android/release/deps/libmagnum_opus-74c3b5e4640ff11a.rlib(opus_decoder.o) is incompatible with aarch64linux
          ld: error: too many errors emitted, stopping now (use -error-limit=0 to see all errors)
          clang: error: linker command failed with exit code 1 (use -v to see invocation)

how to resolve .

**Comment by basilgello** [comment, marked as answer]
For a full build process on Android, refer to https://github.com/rustdesk/rustdesk/blob/master/flutter/build_fdroid.sh

The Debian dependencies indtallable by `apt-get` are here: https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/com.carriez.flutter_hbb.yml#L855

**Comment by rockjl** [comment]
This issue is caused by some libraries not being installed in the system.
Execute this command to install the necessary dependency: sudo E/ flutter/build_android_deps.sh arm64-v8a
Finally, come and execute:/ Flutter/ndk_arm64.sh will pass

**Reply by aaawwwmmm** [reply]
How to solve the same problem I am also encountering

**Comment by aaawwwmmm** [comment]
how to salve the same problem
